File: online-game-store-api/OnlineGameStore.DAL/Entities/Comment.cs
using System;
using System.Collections.Generic;

namespace OnlineGameStore.DAL.Entities
{
    public class Comment : BaseEntity
    {
        public string Name { get; set; }
        public string Body { get; set; }
        public bool IsQuoted { get; set; }

        public Guid GameId { get; set; }
        public virtual Game Game { get; set; }

        public Guid? ParentCommentId { get; set; }
        public virtual Comment ParentComment { get; set; }

        public virtual ICollection<Comment> ChildComments { get; set; }
    }
}

File: online-game-store-front/src/app/classes/comment.ts
export class Comment {
  [prop: string]: any;
  constructor(
  public id: string,
  public name: string,
  public body: string | null,
  public isQuoted: boolean,
  public gameId: string | null,
  public parentCommentId: string | null,
  public parentCommentName: string | null,
  public parentCommentBody: string | null,
  public childComments: Comment[] | null)
  {

  }
}